src/corehost/corehost.cpp

```diff
@@ -112,13 +112,13 @@ int exe_start(const int argc, const pal::char_t* argv[])
         requires_v2_hostfxr_interface = true;
     }
 
-    bundle::bundle_runner_t extractor(host_path);
-    StatusCode bundle_status = extractor.extract();
+    bundle::runner_t::bundle_path = host_path;
```

Comment:
I personally would prefer the runner to be an instance. And then pass the pass as an argument to the extract... setting a field like this doesn't feel right.

Reply:
https://github.com/dotnet/core-setup/pull/6576#discussion_r289160318